주문 기능 구현

tlqkrus012345 opened the pull request.

tlqkrus012345 committed:
refactor: url

Resolves: feature/19

tlqkrus012345 committed:
feat: 주문, 주문상품 도메인

Resolves: #19

tlqkrus012345 committed:
feat: 주문 기능

Resolves: #19

tlqkrus012345 committed:
feat: 주문 컨트롤러

Resolves: #19

tlqkrus012345 committed:
test: 주문 서비스, 컨트롤러

Resolves: #19

tlqkrus012345 committed:
feat: 주문 http

Resolves: #19

f-lab-martin commented on http/order.http:
```diff
@@ -0,0 +1,10 @@
+  "productQuantityMap": {
+    "1": 2,
+    "2": 4
+  }
```
json 포맷으로 map은 권장되지 않습니다. 각각의 필드가 어떤 의미인지 알기가 어려워요~! key, value가 각각 어떤 의미인지 알게 하려면 어떻게 바꾸는 것이 좋을까요?

f-lab-martin commented on src/main/java/com/paytogether/order/service/OrderService.java:
```diff
@@ -0,0 +1,99 @@
+      canOrderProducts.add(productId);
+    }
+    return canOrderProducts;
+  }
```
일급 클래스에 대해 알아보시고 적용해보시면 어떨까요?
그리고 for loop이 아닌 stream을 이용하면 코드가 어떻게 달라질까요?

f-lab-martin commented on src/test/java/com/paytogether/order/controller/OrderControllerTest.java:
```diff
@@ -0,0 +1,89 @@
+    given(loginCheckInterceptor.preHandle(any(), any(), any())).willReturn(true);
+  }
+
+  @DisplayName("주문 성공 테스")
```
오타가 있는 것 같습니다.
그리고 실패하는 테스트케이스도 있으면 어떨까요?

f-lab-martin commented on src/test/java/com/paytogether/order/service/OrderServiceTest.java:
```diff
@@ -0,0 +1,120 @@
+        .containsExactlyInAnyOrder(
+            tuple(productId1, expectedStock1), // 10 - 5
+            tuple(productId2, expectedStock2) // 20 - 10
+        );
```
어떤 테스트인지 가독성 좋게 정리하려면 어떻게 하는 게 좋을까요?

tlqkrus012345 committed:
refactor: order 서비스 facade 적용

Resolves: #19

tlqkrus012345 committed:
test: order 서비스 테스트

Resolves: #19

tlqkrus012345 committed:
feat: 재고 감소 기능

Resolves: #19

tlqkrus012345 committed:
feat: 상품 조회 기능

Resolves: #19

tlqkrus012345 committed:
feat: orderproduct 생성 기능

Resolves: #19

tlqkrus012345 committed:
feat: ProductQuantity 일급클래스

- 상품 아이디와 수량을 Map을 통해 보관하는 것은 키와 벨류 값을 인지하기 어렵다 판단하여 적용
- ProductQuantity와 관련된 행위를 하나의 클래스에서 관리하기 위해 적용

Resolves: #19

tlqkrus012345 committed:
refactor: 파라미터

- Map -> List<>

Resolves: #19

tlqkrus012345 committed:
refactor: json

- map -> List<>

Resolves: #19

tlqkrus012345 committed:
build: 테스트 롬북

Resolves: #19

tlqkrus012345 committed:
test: 회원로그인 헬퍼 클래스

- 중복 코드 줄이기 위해 적용

Resolves: #19

tlqkrus012345 committed:
refactor: id -> productid

Resolves: #19

tlqkrus012345 committed:
test: 재고 감소

Resolves: #19

tlqkrus012345 committed:
test: 주문 컨트롤러

- requestbody. map -> list

Resolves: #19

tlqkrus012345 committed:
refactor: 등록되지 않은 상품 필터 기능

- 기존에 List<Product>을 ProductQUantities에서 업데이트 후 사용하여 주문 가능한 상품 한 곳에서 관리

Resolves: #19

tlqkrus012345 committed:
refactor: 상품 가격 ProductQuantities 클래스로 이동

Resolves: #19

tlqkrus012345 committed:
test: 등록되지 않은 상품 주문 테스트

Resolves: #19

tlqkrus012345 committed:
test: 재고 감소 실패 테스트

Resolves: #19

f-lab-martin reviewed (approved):
코멘트 반영한 부분 모두 확인했습니다~!
새로 추가된 클래스의 경우에는 getter, setter를 제외한 별도 로직이 있다면 최대한 테스트코드를 작성해주시면 좋을 것 같습니다. 이건 다음 PR에 반영해주세요!
